### ISSUE TITLE
Alert users of modified SK placed snippets when visiting module settings page

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#2817]
## Feature Description
The Analytics and AdSense module (and potentially others) are designed to add a script tag to the site front end (unless they detect an existing tag). Sometimes however, a site does not properly display the tags - for example if the site's theme does not fire the `wp_head` tag, or some optimization/concatinating/caching plugins that adjust the `wp_head` content. In these cases, the tag will not be output correctly (or the output might be delayed) and the site will not collect data (Analytics) or show ads (AdSense).

We can improve this by using our existing script detection mechanism to verify Site Kit's tag insertion and warn users when a tag is not successfully inserted.

For Analytics setup: after the user completes the setup, if they have chosen to insert the snippet, we can verify that the snippet is being successfully inserted - if not, we can warn the user and potentially block them from completing setup.

For AdSense setup, once the user connects to their AdSense account, we begin inserting the tag. We can begin verification checks here as well and warn the user if the tag isn't being placed successfully.

Settings screen - when user's go to check their module settings, we can perform the tag checks and warn the users if tags are not being inserted.

## Original Description

There have been multiple support topics open in relation to users Analytics tags not firing. From troubleshooting efforts many users have optimization plugins active with settings such as minification and combining of assets. This in turn resulted in no visits being tracked with the Site Kit placed Analytics snippet. Many of these optimisation plugins do feature warnings, although not all. 

A valid snippet status could be output in the "_Site Kit > settings > 'module' > Edit_" screen. Similar to the pr-setup checks it could alert the user to a "non standard Site Kit placed snippet". 

Example support topics with Analytics not firing due to optimization plugin interference: 
- https://wordpress.org/support/topic/tag-manager-not-working/#post-13808938
- https://wordpress.org/support/topic/log-in-user-does-not-excluded-from-analytics/
- https://wordpress.org/support/topic/google-analytics-sitekit-has-suddenly-stopped-working/
- https://wordpress.org/support/topic/tag-manager-not-working/
- https://wordpress.org/support/topic/log-in-user-does-not-excluded-from-analytics/
---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by jamesozzie** [comment]
@adamsilverstein Similar to the Analytics pre-activation checks is it possible a check could be performed on the edit screen to alert users of a non standard or modified snippet output?

**Comment by adamsilverstein** [comment]
> Similar to the Analytics pre-activation checks is it possible a check could be performed on the edit screen to alert users of a non standard or modified snippet output?

Yes, I think we could do a check on the front end to see if the snippet is being placed as expected, then use the results to help the user.

**Comment by adamsilverstein** [comment]
@jamesozzie I updated the title/description to describe the overall problem and a potential solution; I left your original description just below that. Moving to prioritization.

@felixarntz this ticket proposes adding detection of our own tags being placed when enabled, and warning users when there is an issue.

**Comment by marrrmarrr** [comment]
This is definitely worth implementing, however it will require some UX work (we need an extra step to check and display the results to the users). cc @aarondruck - let's discuss. 

@adamsilverstein from the description you added, it sounds like we would be able to check if tags are correctly placed before the user clicks "Configure Analytics" -- is that accurate? When is the tag actually placed on the page, once the user selects all dropdowns for account/property/etc, or after they click "configure Analytics"?  This is important to clarify so we know at what step to display the new UI.

**Comment by felixarntz** [comment]
@marrrmarrr I don't think this is ready for acceptance criteria yet given your above comment, so let's move this back to "Backlog". We should only be moving items to acceptance criteria when the work is mostly scoped and just has to be added to the issue (i.e. issues shouldn't typically sit in "Acceptance Criteria" for more than 1-2 weeks or so).

**Comment by adamsilverstein** [comment]
>  it sounds like we would be able to check if tags are correctly placed before the user clicks "Configure Analytics" -- is that accurate?

@marrrmarrr Right. We could add the check even before the user submits the form (on activation or in settings). 

We would do something like request the home page with a special query parameter that causes a test tag to be inserted only for that page load. The check would verify that process works - essentially testing that the active theme properly fires the `wp_head` action. 

If the test tag doesn't appear as expected we can inform the user they may have an issue that will prevent tag insertion from working.

**Comment by marrrmarrr** [comment]
> We should only be moving items to acceptance criteria when the work is mostly scoped and just has to be added to the issue 
Thanks @felixarntz, my thought here was that we've decided to take action on this, so it should not be left in the backlog. I'll make a note for the future. 

@adamsilverstein from what you're suggesting it sounds like we can do this check in the UI before they hit submit and display something along the lines of the [compatibility checks](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/setup-using-proxy--start-%C2%A0with-error) we show on setup, but maybe with this [visual treatment](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/global-notices--settings-warning-notice).

**Comment by aaemnnosttv** [comment]
Closing as this was largely addressed by #5695 which adds detected tag status checks to the Site Health "Status" tab.
